include/dbus_privileges.hpp

```diff
@@ -49,10 +51,12 @@ inline bool
     session.userRole = userRole;
     BMCWEB_LOG_DEBUG("userName = {} userRole = {}", session.username, userRole);
 
+    session.isGenerateSecretKeyRequired = secretKeyRequired.value_or(false);
     // Set isConfigureSelfOnly based on D-Bus results.  This
     // ignores the results from both pamAuthenticateUser and the
     // value from any previous use of this session.
-    session.isConfigureSelfOnly = passwordExpired.value_or(false);
+    session.isConfigureSelfOnly = passwordExpired.value_or(false) ||
+                                  session.isGenerateSecretKeyRequired;
```

Comment:
passwordExpired.value_or(false) || secretKeyRequired.value_or(false)  ?

Reply:
why are you using session.isGenerateSecretKeyRequired?